clippy_lints/src/eta_reduction.rs

```diff
@@ -101,19 +101,20 @@ fn check_clousure<'tcx>(cx: &LateContext<'tcx>, outer_receiver: Option<&Expr<'tc
     };
 
     if body.value.span.from_expansion() {
-        if body.params.is_empty() {
-            if let Some(VecArgs::Vec(&[])) = VecArgs::hir(cx, body.value) {
-                // replace `|| vec![]` with `Vec::new`
-                span_lint_and_sugg(
-                    cx,
-                    REDUNDANT_CLOSURE,
-                    expr.span,
-                    "redundant closure",
-                    "replace the closure with `Vec::new`",
-                    "std::vec::Vec::new".into(),
-                    Applicability::MachineApplicable,
-                );
-            }
+        if body.params.is_empty() &&let Some(VecArgs::Vec(&[])) = VecArgs::hir(cx, body.value)
+            // FIXME: When we can correctly determine if `alloc` crate can use in a `no_std` environment, we suggest using `alloc::vec::Vec::new` here.
+            && let Some("std")=std_or_core(cx)
```

Comment:
The newly added condition and the existing condition are combined in one `if` statement to avoid being linted by the `collapsible_if`.